cirq/ops/linear_combinations.py

```diff
@@ -500,17 +500,50 @@ def __neg__(self):
         factory = type(self)
         return factory(-self._linear_dict)
 
-    def __imul__(self, a: value.Scalar):
-        self._linear_dict *= a
+    def __imul__(self, other: PauliSumLike):
+        if not isinstance(other, (float, int, complex, PauliString, PauliSum)):
```

Comment:
Use `numbers.Complex` instead of ` float, int, complex`.

Reply:
done